homeassistant/components/teslemetry/binary_sensor.py

```diff
@@ -0,0 +1,272 @@
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TeslemetryBinarySensorEntityDescription(
+        key="vehicle_state_pf",
+        device_class=BinarySensorDeviceClass.DOOR,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TeslemetryBinarySensorEntityDescription(
+        key="vehicle_state_pr",
+        device_class=BinarySensorDeviceClass.DOOR,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+ENERGY_LIVE_DESCRIPTIONS: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(key="backup_capable"),
+    BinarySensorEntityDescription(key="grid_services_active"),
+)
+
+
+ENERGY_INFO_DESCRIPTIONS: tuple[BinarySensorEntityDescription, ...] = (
+    BinarySensorEntityDescription(
+        key="components_grid_services_enabled",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Teslemetry binary sensor platform from a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        chain(
+            (  # Vehicles
+                TeslemetryVehicleBinarySensorEntity(vehicle, description)
+                for vehicle in data.vehicles
+                for description in VEHICLE_DESCRIPTIONS
+            ),
+            (  # Energy Site Live
+                TeslemetryEnergyLiveBinarySensorEntity(energysite, description)
+                for energysite in data.energysites
+                for description in ENERGY_LIVE_DESCRIPTIONS
+                if energysite.info_coordinator.data.get("components_battery")
+            ),
+            (  # Energy Site Info
+                TeslemetryEnergyInfoBinarySensorEntity(energysite, description)
+                for energysite in data.energysites
+                for description in ENERGY_INFO_DESCRIPTIONS
+                if energysite.info_coordinator.data.get("components_battery")
+            ),
+        )
+    )
+
+
+class TeslemetryVehicleBinarySensorEntity(TeslemetryVehicleEntity, BinarySensorEntity):
+    """Base class for Teslemetry vehicle binary sensors."""
+
+    entity_description: TeslemetryBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        data: TeslemetryVehicleData,
+        description: TeslemetryBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = description
+        super().__init__(data, description.key)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the binary sensor."""
+
+        if self.coordinator.updated_once:
+            if self._value is None:
+                self._attr_available = False
+                self._attr_is_on = None
+            else:
+                self._attr_available = True
+                self._attr_is_on = self.entity_description.is_on(self._value)
+        else:
+            self._attr_is_on = None
```

Comment:
Should we be setting attr_available to false in this branch?

Reply:
No, because in this condition the value is unknown but is potentially available.

Specifically this is when the vehicle is asleep while the integration starts.